### PR TITLE
Update Tensor Mechanics to new static validParams

### PR DESCRIPTION
**Pull request by cticenhour**
Refs #14135 

@sapitts @tophmatthews @bwspenc FYI This should not break any existing code or functionality (the old validParams function implementation returns the new static implementation) while we transition elsewhere. If you have any questions or concerns, direct them to me, @aeslaughter, or @rwcarlsen.

### CODE REVIEW

**Comment by aeslaughter** [comment]
This failed the unity build, which I couldn't reproduce on MacOS. I did find a template parameter that didn't get updated, but that is likely un-related.

**Comment by cticenhour** [comment]
There were several instances where the old version of the code called a particular version of the template, rather than leaving it as `compute_stage`. I had thought that was intentional, so left it when I saw it.

**Comment by cticenhour** [comment]
FYI - After a small test of moving the definitions of `validParams` and `decompositionType` into the header for `ADComputeFiniteStrain`, the non unity build completed successfully. Something weird is going on with how things are being included in this object...but nothing looks like an obvious difference from other objects. I'll hopefully track this down this afternoon.

**Comment by aeslaughter** [comment]
@cticenhour Thanks for digging into this, I messed with it this weekend a bit and was confused enough to give up.

**Comment by cticenhour** [comment]
Turns out the issue was with the `defineADLegacyParams` macro.

In that macro, we had:
```
  template <>                                                                                      \
  InputParameters validParams<ADObjectType<JACOBIAN>>()                                            \
  {                                                                                                \
    return ADObjectType<RESIDUAL>::validParams();                                                  \
  } 
```

Changing the return template parameter here to `JACOBIAN` for more consistency both corrected the compiler error in Linux (on a rod duplication of the testing environment) and didn't cause further issues in the rest of the framework (at least as far as my local testing goes). Not sure why the original macro caused trouble only in `ADComputeFiniteStrain`, but I'll take this fix. 😄

**Comment by rwcarlsen** [comment]
The original defineADValidParams macro works that way - always calls with the RESIDUAL template arg.  But it doesn't work for the new member func validParams because of something to do with the fact that calling it now instantiates the class template instead of instantiating a global function template.  I still don't understand why there would be a problem, but at least we discovered something that is a convincing cause.

**Review by rwcarlsen — APPROVED** [review]

File: modules/tensor_mechanics/include/vectorpostprocessors/LineMaterialRankTwoSampler.h [hunk 1]
```diff
@@ -30,6 +30,8 @@ class LineMaterialRankTwoSampler : public LineMaterialSamplerBase<RankTwoTensor>
    * Sets up variables for output based on the properties to be output
    * @param parameters The input parameters
    */
+  static InputParameters validParams();
```
Comment (bwspenc) [inline modules/tensor_mechanics/include/vectorpostprocessors/LineMaterialRankTwoSampler.h]: @cticenhour I just noticed today that the new method got put between the constructor and the description of the constructor, making the comments above this invalid.  There are probably a number of these. Would you mind checking for cases like this and fixing the ones that are affected?

Comment (cticenhour) [reply modules/tensor_mechanics/include/vectorpostprocessors/LineMaterialRankTwoSampler.h]: Thanks for the catch - I believe this was the result of a script that moved this declaration to a particular location in the file (referenced to the position of the constructor). I'll put in a PR soon fixing this where I find it.

**Pull request merged**